### PR TITLE
feat: Introduce build matrix for kube versions

### PR DESCRIPTION
**Pull request by bradbeam**
Signed-off-by: Brad Beam <brad.beam@carbonrelay.com>

### CODE REVIEW

File: .github/workflows/kubematrix.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
+name: Main workflow
```
Comment (jgustie) [inline .github/workflows/kubematrix.yaml]: Definitely change the name, I think it's confusing the branch restrictions.

**Comment by jgustie** [comment]
I wonder if we can pull the output of a previous build instead of having to rebuild just for this workflow?

**Comment by bradbeam** [comment]
> I wonder if we can pull the output of a previous build instead of having to rebuild just for this workflow?

I think the only way to make this happen is to have the jobs be part of the same workflow. We can add in the matrix build stuff into the master build if you want to go that route. Since it runs on push to master I'm not sure I mind too much on any additional time for this, but I can see the benefit of using the exact same artifacts that would be published.

**Comment by jgustie** [comment]
This looks like it's going to attempt to re-release the product since it's re-deploying everything and it's running on a tag build.

**Comment by bradbeam** [comment]
Should we merge the workflows then?

**Comment by jgustie** [comment]
At a minimum we just shouldn't trigger this workflow on tag pushes (in theory it would have already run on the commit push to master anyway).

**Review by jgustie — APPROVED** [review]

**Pull request merged**